lib/v8debugapi.js

```diff
@@ -384,10 +384,25 @@ module.exports.create = function(logger_, config_, fileStats_) {
   // }
 
   function findScripts(scriptPath) {
+    // Use repository relative mapping if present
+    if (config.appPathRelativeToRepository) {
+      var candidate = scriptPath.replace(config.appPathRelativeToRepository,
+        config.workingDirectory);
+      // There should be no ambiguity resolution if project root is provided
+      return fileStats[candidate] ? [ candidate ] : [];
+    }
+    // Next try to match path
     var regexp = pathToRegExp(scriptPath);
     var matches = Object.keys(fileStats).filter(function(script) {
       return regexp.test(script);
     });
+    // Finally look for files with the same name regardless of path
+    if (matches.length === 0) {
+      scriptPath = path.basename(scriptPath);
```

Comment:
We now do a reverse search through the path components refining a list of candidates each iteration to find the right file in this situation.